packages/keyring-controller/jest.environment.js

```diff
@@ -0,0 +1,16 @@
+const NodeEnvironment = require('jest-environment-node');
+
+/**
+ * KeyringController depends on @noble/hashes, which as of 1.3.2 relies on the
+ * Web Crypto API in Node and browsers.
+ */
+class CustomTestEnvironment extends NodeEnvironment {
+  async setup() {
+    await super.setup();
+    if (typeof this.global.crypto === 'undefined') {
+      this.global.crypto = require('crypto').webcrypto;
```

Comment:
`@noble/hashes` will use Web Crypto API within a Node context. Specifically, [this file](https://github.com/paulmillr/noble-hashes/blob/1.3.2/src/cryptoNode.ts) will get loaded thanks to [this line](https://github.com/paulmillr/noble-hashes/blob/1.3.2/package.json#L55) in `package.json`.

Unfortunately, that's not happening in the tests, and [this file](https://github.com/paulmillr/noble-hashes/blob/1.3.2/src/crypto.ts) gets loaded instead, which attempts to read `global.crypto` instead of `require('crypto').webcrypto`. I'm not 100% sure on this, but I believe that's happening because of the way we've configured TypeScript: [we're using a `moduleResolution` of `node`](https://github.com/MetaMask/core/blob/4e8558e33bd5fe9f1861a40741ea23d6a9b02747/tsconfig.packages.json#L7), i.e. "classic" mode, which means that TypeScript skips the "exports" field in `package.json`.